### PR TITLE
Fix flycapture install when using local copy

### PR DESCRIPTION
**Pull request by tspicer01**
The local install of the libflycapture is not resolving the real file name and instead is installing the symlink.
This results in a broken symlink.

We recently switched from downloading the library on each build to having the library locally installed which is what triggered this change.  

The fix will copy the real library into the build space, resolving the symlinks and use the copy to install into the bundle.

### CODE REVIEW

**Comment by mikepurvis** [comment]
LGTM. I think this code dates from before the deb installed a symlink like that.

**Review by mikepurvis — APPROVED** [review]

File: pointgrey_camera_driver/CMakeLists.txt [hunk 1]
```diff
@@ -19,14 +19,21 @@ catkin_package(
...
 # website during this build process.
-find_library(POINTGREY_LIB flycapture)
+find_library(POINTGREY_LIB NAMES libflycapture.so.2 flycapture)
 if(NOT POINTGREY_LIB)
   message(STATUS "libflycapture not found in system library path")
   include(cmake/DownloadFlyCap.cmake)
   download_flycap(POINTGREY_LIB POINTGREY_INCLUDE_DIR) 
   message(STATUS "libflycapture library: ${POINTGREY_LIB}")
   message(STATUS "libflycapture include: ${POINTGREY_INCLUDE_DIR}")
   include_directories(${POINTGREY_INCLUDE_DIR})
```
Comment (efernandez) [inline pointgrey_camera_driver/CMakeLists.txt:29]: I wonder why this isn't needed when the library is found. I guess the default path for the headers does the job, but maybe we should still figure out how to set `POINTGREY_INCLUDE_DIR` properly. ?!

File: pointgrey_camera_driver/CMakeLists.txt [hunk 2]
```diff
@@ -19,14 +19,21 @@ catkin_package(
...
 if(NOT POINTGREY_LIB)
   message(STATUS "libflycapture not found in system library path")
   include(cmake/DownloadFlyCap.cmake)
   download_flycap(POINTGREY_LIB POINTGREY_INCLUDE_DIR) 
   message(STATUS "libflycapture library: ${POINTGREY_LIB}")
   message(STATUS "libflycapture include: ${POINTGREY_INCLUDE_DIR}")
   include_directories(${POINTGREY_INCLUDE_DIR})  
+else()
+  # flycapture exists on the machine copy it into the build directory/resolving all symlinks
+  message(STATUS "libflycapture library found on system: ${POINTGREY_LIB}")
```
Comment (efernandez) [inline pointgrey_camera_driver/CMakeLists.txt]: pedantic: I'd make the messages more like the ones on the other cases.

``` cmake
message(STATUS "libflycapture found in system library path")
message(STATUS "libflycapture library: ${POINTGREY_LIB}")
```

File: pointgrey_camera_driver/CMakeLists.txt [hunk 3]
```diff
@@ -19,14 +19,21 @@ catkin_package(
...
+  GET_FILENAME_COMPONENT(REAL_FLYCAPTURE ${POINTGREY_LIB} REALPATH)
+  configure_file(${REAL_FLYCAPTURE} ${CMAKE_CURRENT_BINARY_DIR}/${POINTGREY_LIB} COPYONLY)  
+  set(POINTGREY_LIB ${CMAKE_CURRENT_BINARY_DIR}/${POINTGREY_LIB})
+  message(STATUS "Flycapture lib copied from : ${REAL_FLYCAPTURE} into ${POINTGREY_LIB}")
```
Comment (efernandez) [inline pointgrey_camera_driver/CMakeLists.txt]: pedantic: also here.

``` cmake
message(STATUS "libflycapture copied from ${REAL_FLYCAPTURE} into ${POINTGREY_LIB}")
```

**Comment by efernandez** [comment]
Only pedantic comments re. the messages.

Tested and works fine now, for a bundle with the broken link. But the new cmake doesn't use it at all from what I see when I run `cmake .` on the build folder created by `catkin build`:

``` bash
-- libflycapture library found on system: /usr/lib/libflycapture.so.2
-- Flycapture lib copied from : /usr/lib/libflycapture.so.2.9.3.43 into ~/ws/build/pointgrey_camera_driver//usr/lib/libflycapture.so.2
```

@tspicer01 Would this pick the library from the bundle instead, if it's there and the link isn't broken?
It probably doesn't matter.

:+1:

**Review by efernandez — APPROVED** [review]

File: pointgrey_camera_driver/CMakeLists.txt (hunk 1 above)
Comment (tspicer01) [reply pointgrey_camera_driver/CMakeLists.txt:29]: What do you need POINTGREY_INCLUDE_DIR for?
It's only used to set the makefile include path to add the non-standard location when we download it.  It's not needed if the library is found b/c the headers are already in the standard location, so the include path is already included in the compile.
If we were redistributing the header files like we are doing with the library then yes we would need to know their location.

File: pointgrey_camera_driver/CMakeLists.txt (hunk 2 above)
Comment (tspicer01) [reply pointgrey_camera_driver/CMakeLists.txt]: I've fixed the messages to be more standard to what was already there.

**Comment by tspicer01** [comment]
To answer your question @efernandez 
Once the bundle is fixed it can pull it from there first.  I "fixed" my bundle and this is what you would see:
-- libflycapture found in system library path
-- libflycapture library: /opt/clearpath/2.4devel/sdk/lib/libflycapture.so.2
-- libflycapture copied from : /opt/clearpath/2.4devel/sdk/lib/libflycapture.so.2 into /home/tspicer/dependencies/flycapture/build/pointgrey_camera_driver/pointgrey_camera_driver//opt/clearpath/2.4devel/sdk/lib/libflycapture.so.2

The install directory will still be:
install/lib/libflycapture.so.2

**Pull request merged**